src/components/AddressLink/index.tsx

```diff
@@ -16,31 +17,75 @@ const Link = styled(SmartLink)`
 `
 
 export const AddressLink = ({
-  label,
+  id,
   address,
-  urlPrefix
+  label,
+  disableLink,
+  stopPropagation,
+  hideCopy,
+  hideCopyText
 }: AddressLinkProps) => {
+  const truncated = useMemo(
+    () => (address ? truncateAddress(address) : ''),
+    [address]
+  )
+  const explorerUrl =
+    process.env.NEXT_PUBLIC_EXPLORER_URL ||
+    'https://explorer-calibration.glif.link'
+  const linkText = truncated && id ? `(${id})` : id || truncated
+  const linkHref = `${explorerUrl}/actor/?address=${id || address}`
+  const onClick = (e: MouseEvent) => stopPropagation && e.stopPropagation()
```

Comment:
Should we make this a `useCallback` just for shits and gigs while we're in a review?

Reply:
I have no beef with that